src/Tribe/Admin/Notice/Install_Event_Tickets.php

```diff
@@ -0,0 +1,436 @@
+
+use WP_Upgrader;
+use WP_Ajax_Upgrader_Skin;
+use Plugin_Upgrader;
+use Tribe__Main;
+use Tribe__Admin__Notices;
+use Tribe__Template;
+
+/**
+
+ */
+class Install_Event_Tickets {
+
+	/**
+	 * Stores the plugin slug.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected static $plugin_slug = 'event-tickets';
+
+	/**
+	 * Stores the assets group ID for the notice.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected static $assets_group = 'tribe-events-admin-notice-install-event-tickets';
+
+	/**
+	 * Stores the instance of the notice template.
+	 *
+	 * @since TBD
+	 *
+	 * @var Tribe__Template
+	 */
+	protected $template;
+
+	/**
+	 * Register update notices.
+	 *
+	 * @since TBD
+	 */
+	public function hook() {
+		if ( ! is_admin() || ! current_user_can( 'install_plugins' ) ) {
+			return;
+		}
+
+		add_action( 'wp_ajax_notice_install_event_tickets', [ $this, 'ajax_handle_notice_install_event_tickets' ] );
+
+		$this->assets();
+		$this->notice_install();
+		$this->notice_activate();
+	}
+
+	/**
+	 * Register `Install` notice assets.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function assets() {
+		$plugin = tribe( 'tec.main' );
+
+		tribe_asset(
+			$plugin,
+			'tribe-events-admin-notice-install-event-tickets-js',
+			'admin/notice-install-event-tickets.js',
+			[
+				'jquery',
+				'tribe-common',
+			],
+			null,
+			[
+				'localize' => [
+					[
+						'name' => 'TribeEventsAdminNoticeInstall',
+						'data' => [ 'ajaxurl' => admin_url( 'admin-ajax.php', ( is_ssl() ? 'https' : 'http' ) ), ]
+					],
+				],
+				'groups'   => [
+					self::$assets_group,
+				],
+			]
+		);
+
+		tribe_asset(
+			$plugin,
+			'tribe-events-admin-notice-install-event-tickets-css',
+			'admin/notice-install-event-tickets.css',
+			[ 'tec-variables-full' ],
+			[
+				'admin_enqueue_scripts',
+				'wp_enqueue_scripts',
+			],
+			[
+				'groups' => self::$assets_group,
+			]
+		);
+	}
+
+	/**
+	 * Get the plugin path for `Event Tickets`, by default.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $slug The plugin slug.
+	 *
+	 * @return string $path The plugin path.
+	 */
+	protected function get_plugin_path( $slug = '' ): string {
+		if ( empty( $slug ) ) {
+			$slug = self::$plugin_slug;
+		}
+
+		return $slug . '/' . $slug . '.php';
+	}
+
+	/**
+	 * Checks if `Event Tickets` is installed.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $slug The plugin slug.
+	 *
+	 * @return boolean True if active
+	 */
+	public function is_installed( $slug = '' ): bool {
+		$installed_plugins = get_plugins();
+
+		return array_key_exists( $this->get_plugin_path( $slug ), $installed_plugins ) || in_array( $this->get_plugin_path( $slug ), $installed_plugins, true );
+	}
+
+	/**
+	 * Checks if `Event Tickets` is active.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $slug The plugin slug.
+	 *
+	 * @return boolean True if active.
+	 */
+	public function is_active( $slug = '' ): bool {
+		return is_plugin_active( $this->get_plugin_path( $slug ) );
```

Comment:
This check can lead to false negatives. Plugins can be installed in any path that we choose. We should be checking for `! did_action( 'tribe_tickets_plugin_loaded' )`

Reply:
I'm relying on a WP core function, that's used to basically check if a plugin is active. I John Doe is renaming the `event-tickets` (in this case) folder to something else then it should be their problem.

Reply:
You are not wrong, but PUE shouldn't be the standard; it's an older code, honestly. I agree with you that it would be amazing to use WordPress method here, but it's slower and if you have ET active but not in use example; If there is a dependency with TEC failing, it would show the notice in this case and we want to avoid.

Reply:
@bordoni This notice lives in TEC, and is for installing ET. 

The notice will never be shown if there's a dependency mismatch because both plugins will be active. And in that case, the notice is not shown (that's what this check is for). `is_plugin_active` will be the main reason why it won't be shown.

In my opinion, WP core functions are the way to go probably 99% of the time. I don't feel using [this code](https://github.com/WordPress/wordpress-develop/blob/6.1/src/wp-admin/includes/plugin.php#L527-L529) to perform the check will make things significantly slower. This is the code for `is_plugin_active`:

`function is_plugin_active( $plugin ) {
	return in_array( $plugin, (array) get_option( 'active_plugins', array() ), true ) || is_plugin_active_for_network( $plugin );
}`